src/burrow/burrow.js

```diff
@@ -38,12 +39,40 @@ class Burrow extends BlockchainInterface{
     }
 
     /**
-     * Deploy the chaincode specified in the network configuration file to all peers.
-     * @return {Burrow} resolve
+     * Deploy the smart contract specified in the network configuration file.
+     * @return {object} Promise execution for namereg.
      */
-    installSmartContract() {
-        // TODO: complete
-        return Promise.resolve();
+    async installSmartContract() {
+        let config  = require(this.configPath);
+        let grpc = config.burrow.network.validator.grpc;
+        if(grpc === null) {
+            logger.error('Error: Validator url not set.');
+        }
+        let account = fs.readFileSync(util.resolvePath(config.burrow.network.validator.address)).toString();
+        let options = {objectReturn: true};
+        let burrow = monax.createInstance(grpc, account, options);
+        let data = JSON.parse(fs.readFileSync(util.resolvePath(config.contract.path)).toString());
+        let abi = data.Abi;
+        let bytecode = data.Evm.Bytecode.Object;
+
+        const contract = burrow.contracts.new(abi, bytecode);
+        let contractAddress = await Promise.all([contract._constructor('')]).then(([address]) => {
```

Comment:
My suggestion to avoid Promise chains:
```js
let contractAddress = await contract._constructor('');
logger.infor(`Contract: ${address}`);
```
Continued in the next comment.